distributed/diagnostics/bokeh/status/server_lifecycle.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+
+from __future__ import print_function, division, absolute_import
+
+from collections import deque
+from datetime import datetime
+import json
+
+from tornado import gen
+from tornado.locks import Condition
+from tornado.httpclient import AsyncHTTPClient
+
+import distributed.diagnostics
+from distributed.utils import log_errors
+
+client = AsyncHTTPClient()
+
+messages = {}  # Globally visible store of messages
+distributed.diagnostics.messages = messages  # monkey-patching
+
+
+@gen.coroutine
+def http_get(route):
+    """ Get data from JSON route, store in messages deques """
+    with log_errors():
+        response = yield client.fetch('http://localhost:9786/%s.json' % route)
+        msg = json.loads(response.body.decode())
+        messages[route]['deque'].append(msg)
+        messages[route]['condition'].notify_all()
+        messages[route]['times'].append(datetime.now())
+
+
+def on_server_loaded(server_context):
+    messages['workers'] = {'interval': 1000,
+                           'deque': deque(maxlen=1000),
+                           'times': deque(maxlen=1000),
+                           'condition': Condition()}
+    server_context.add_periodic_callback(lambda: http_get('workers'), 1000)
+
+    messages['tasks'] = {'interval': 100,
+                         'deque': deque(maxlen=1000),
+                         'times': deque(maxlen=1000),
+                         'condition': Condition()}
+    server_context.add_periodic_callback(lambda: http_get('tasks'), 100)
```

Comment:
@canavandl 

OK, so now we grab the messages from the scheduler on a per-process basis, rather than a per-session/tab basis.  Moreover, we record the last thousand messages we've seen from the scheduler so that, if necessary, plots can query message history.

Now, rather than waiting on a `client.fetch` future, code should wait on `condition.wait()` for their particular message stream.  They'll be woken up every time a new message comes in.